locator/tablets.cc

```diff
@@ -700,6 +722,71 @@ class tablet_effective_replication_map : public effective_replication_map {
             , _sharder(*_tmptr, table)
     { }
 
+    // Populate the `_tablet_to_dc_replication_factor_map` with replication factors for each data center.
+    // For each tablet index in the tablet map, it retrieves the current replica set using `get_endpoints_for_reading` 
+    // and counts replicas per data center.  Each data center's replication factor (RF) is saved under 
+    // its index according to `_datacenter_map`.
+    seastar::future<> build_dc_replication_factor_map() {
+        const tablet_map& tablets = get_tablet_map();
+        const topology& topo = get_topology();
+        fill_datacenter_map();
+        _tablet_to_dc_replication_factor_map.reserve(tablets.tablet_count());
+        for (size_t tablet_index = 0; tablet_index < tablets.tablet_count(); ++tablet_index) {
+            const tablet_replica_set& replicas = get_endpoints_for_reading(tablet_id{tablet_index});
+            _tablet_to_dc_replication_factor_map.emplace_back();
+            auto& token_rf_list = _tablet_to_dc_replication_factor_map.back();
+            token_rf_list.resize(_datacenter_map.size(), 0);
```

Comment:
There is no such constructor in `small_vector` :(.

Reply:
But I can apply a trick - to use a dummy source so it will use a copy constructor. Not sure tho if it's any faster.

Reply:
My suggestion was mostly stylistic one. It's not mandatory

But how about adding the missing constructor to `small_vector`? There is such a constructor in `std::vector` so having similar one in `small_vector` makes sense.

Reply:
Actually, after performing some tests, I was going to switch from small vector to `absl::inlined_vector`, since for small things it's more economic in terms of memory. Our `small_vector` has an overhead of 24 bytes just for internal pointers. Btw I believe if we replace it for all our small vector replicas, we could get a notifiable memory consumption decrease (as well as better performance).

Reply:
Added a new constructor to small_vector.